agents/check_mk_agent.linux

```diff
@@ -1116,7 +1116,8 @@ run_cached() {
         if [ $((NOW - CF_ATIME)) -ge $((MAXAGE * 2)) ]; then
             # Kill the process still accessing that file in case
             # it is still running. This avoids overlapping processes!
-            fuser -k -9 "$CACHEFILE.new" >/dev/null 2>&1
+            # Also kill all it's subprocesses from the same pgid
+            kill -9 -$(ps -o pgid $(fuser "$CACHEFILE.new" | rev | cut -d':' -f1 | rev ) | tail -1) >/dev/null 2>&1
```

Comment:
Hi!  This pipeline doesn't seem to behave nicely on Debian family distros, so you might like to consider simplifying it e.g. `fuser somefile 2>&1 | awk '{print $2}'` does the trick for me

Reply:
The awk approach will fail upon files with spaces in the name.

Why the 2>&1 after fuser? Let me suggest to redirect 2>/dev/null, then you can also remove rev and cut from the game.

Reply:
Hi!  Apparently I deemed the `awk` approach as acceptable and apparently `2>&1` was required on whatever host I was testing on... back in January.  Things can move really slowly on this repo...

Either way, it doesn't really matter: any approach using `fuser` is obviously going to be non-portable and brittle.  For a while now I have had an idea in mind to fix this once and for all, but at the virtually non-existent rate that submitted code is accepted into the agent scripts, I honestly don't think it's worth the bother :(

Reply:
Now the situation is worse: In a systemd setup, the async processes will be started from a permanently running systemd service, which means *all* jobs will share the same PGID. While that in it self offers a lot of possibilities, we must not rely on that :-(

I am wondering why we don't put the PID of the nohup shell process in the cache files name. Something like
```
echo "...; set -o noclobber; exec > \"${CACHEFILE}.$$.new\" || exit ..." | nohup /bin/bash ...
#                                                  ^^
```
(assuming we get the quotation right). We can then remove all processes by
```
for oldfile in "${CACHEFILE}.*.new"; do 
    kill $(ps -o pid= --ppid $(head -n1 "${oldfile}")
done
```

Either way, I am afraid @SvenVD, your suggestion won't work for all setups.
Do you want to pursue this other approach (or yet another, of course)?
I am asking directly, because I am aware that this PR is around for far too long, so I'd understand if you have other obligations or more fun contributions. In that case I will close this PR.
If you want to keep this going, on the other hand, I will assign myself to it, and do my best to keep it alive.

Reply:
It's been a long time, but I tested my fix on centos 8 which runs systemd and it seemed to work. You could detect running on a RHEL variant and apply the fix there for the time being, better than nothing?

Currently I don't have much time too look into it further

Reply:
We had to change the design for systemd setups quite dramatically this summer, so I'm afraid the tests you carried out are no longer valid :-/
I don't fancy detecting RHEL, because I really think the way to go is to be as portable as possible, and avoid implementing special cases.
I've looked into it over the last days, and I am hoping to kick out `fuser` (and the noclobber shell option as a side effect).
Unfortunately, I don't see any straightforward portable way of killing a process along with all its children (and childrens children!).
Since I can't really see this going anywhere, I will close the issue. Given our current workload, I think it is the most honest thing to do.
However, if anyone comes up with a portable way of doing this, I'm all ears!